### DISCUSSION TITLE
Email to ping

### DISCUSSION THREAD

**Discussion by packerman05** [healthchecks/healthchecks#995] [General]
Hi all.

I'm basically a noob when it comes to this. I followed the instructions from this link.

https://mariushosting.com/how-to-install-healthchecks-on-your-synology-nas/

Everything installs properly as far as I can tell. The issue I'm running into though is the email to ping gives me a local host address in the email to ping address section of the check options. On the healthchecks.io site the link is an actual email address. 

I'm trying to understand what I need to do next to the email to ping setup. 

I want to basically monitor some backup jobs using email notifications. Any guidance on getting this setup would appreciated.

**Comment by cuu508** [comment]
You can configure the domain used in email addresses with the PING_EMAIL_DOMAIN setting: https://healthchecks.io/docs/self_hosted_configuration/#PING_EMAIL_DOMAIN

To start the SMTP listener, you will also need to set the SMTPD_PORT setting: https://github.com/healthchecks/healthchecks/tree/master/docker#smtp-listener-configuration-via-smtpd_port

Potentially useful additional reading here: https://github.com/healthchecks/healthchecks/discussions/970

**Reply by packerman05** [reply]
Thank you for getting back to me. The issue is I don't even know where to start configuring these things. On the top of the 1st link it says Healthchecks prepares its configuration in hc/settings.py. So I assume I need to configure the settings in this file to make changes. I've looked and searched but can't seem to where this file exists let alone edit it.

Also is it not possible to use something like Google smtp to this process?

**Comment by cuu508** [comment]
You are deploying using a Docker image. `hc/settings.py` is baked into the image, so there is not an easy way to make changes to it – you would need to rebuild the image.

But as the next sentence on the top of the 1st link says, `hc/settings.py` reads configuration parameters from environment variables. You can set environment variables in your docker-compose.yml file, in the `environment:` section. If you add `PING_EMAIL_DOMAIN: example.com`, and reload configuration (i.e., update stack? I've not used Portainer...), you should start seeing email addresses ending with example.com. 

> Also is it not possible to use something like Google smtp to this process?

You can use Google SMTP to send email (login links, uptime notifications, reports).

**Comment by cuu508** [comment]
I made tweaks to the documentation in a few places:

* [PING_EMAIL_DOMAIN docs](https://healthchecks.io/docs/self_hosted_configuration/#PING_EMAIL_DOMAIN) now mention the fact you also need `manage.py smtpd` running. It links to a page with instructions on how to use the `SMTPD_PORT` env var
* The [Running with Docker](https://healthchecks.io/docs/self_hosted_docker/) page likewise mentions PING_EMAIL_DOMAIN and links to its docs
* Added a note about SMTPD_PORT in [docker/README.md](https://github.com/healthchecks/healthchecks/tree/master/docker#pre-built-images)

**Comment by packerman05** [comment]
Thanks for getting back to me.

I feel like I'm reading in another language when I try to go through the documentation.

If you were to start fresh and install on a appliance what would be the simplest way to get everything setup from scratch?

Install on Linux? Is there no prepared Docker setups where I can just modify some basic settings to get everything setup?

**Comment by cuu508** [comment]
Here's a walkthrough of setting a Healthchecks instance up on a VPS using docker compose: https://blog.healthchecks.io/2023/05/walk-through-set-up-self-hosted-healthchecks-instance-on-a-vps/

**Comment by scialo** [comment]
Starting SMTP listener on 0.0.0.0:25 ...

```
Traceback (most recent call last):

  File "/opt/healthchecks/./manage.py", line 17, in <module>

    raise e

  File "/opt/healthchecks/./manage.py", line 13, in <module>

    execute_from_command_line(sys.argv)

    ~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^

  File "/usr/local/lib/python3.13/site-packages/django/core/management/__init__.py", line 442, in execute_from_command_line

    utility.execute()

    ~~~~~~~~~~~~~~~^^

  File "/usr/local/lib/python3.13/site-packages/django/core/management/__init__.py", line 436, in execute

    self.fetch_command(subcommand).run_from_argv(self.argv)

    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^

  File "/usr/local/lib/python3.13/site-packages/django/core/management/base.py", line 413, in run_from_argv

    self.execute(*args, **cmd_options)

    ~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^

  File "/usr/local/lib/python3.13/site-packages/django/core/management/base.py", line 459, in execute

    output = self.handle(*args, **options)

  File "/opt/healthchecks/hc/api/management/commands/smtpd.py", line 150, in handle

    controller.start()

    ~~~~~~~~~~~~~~~~^^

  File "/usr/local/lib/python3.13/site-packages/aiosmtpd/controller.py", line 274, in start

    raise self._thread_exception

  File "/usr/local/lib/python3.13/site-packages/aiosmtpd/controller.py", line 235, in _run

    self.server = self.loop.run_until_complete(self.server_coro)

                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^

  File "/usr/local/lib/python3.13/asyncio/base_events.py", line 721, in run_until_complete

    return future.result()

           ~~~~~~~~~~~~~^^

  File "/usr/local/lib/python3.13/asyncio/base_events.py", line 1609, in create_server

    raise OSError(err.errno, msg) from None

PermissionError: [Errno 13] error while attempting to bind on address ('0.0.0.0', 25): [errno 13] permission denied

[uwsgi-daemons] respawning "./manage.py smtpd --port 25 --skip-checks" (uid: 999 gid: 999)
```

**Reply by cuu508** [reply]
* Are you running Healthchecks in docker? 
* What image are you using? 
* What command line are you using to start the container?
* If you use Docker Compose, please provide the docker-compose.yml file

**Comment by scialo** [comment]
i'm running hc in docker.
ver. image docker 3.9.
docker-compose up -d

in my docker-compose:
      -  PING_EMAIL_DOMAIN=mysite.com
      - SMTPD_PORT=25
    ports:
      - 25:25

**Reply by cuu508** [reply]
Try

```
- SMTPD_PORT=2525
ports:
- 25:2525
```

(or anything in the place of 2525 really, as long as it is an unprivileged port)

**Comment by scialo** [comment]
it works!

thanks
